RestaurantReservationSystem.Domain/Services/Interfaces/ITableService.cs

```diff
@@ -0,0 +1,72 @@
+﻿using RestaurantReservationSystem.API.DTOs.Requests;
+using RestaurantReservationSystem.API.DTOs.Responses;
+
+namespace RestaurantReservationSystem.API.Interfaces
+{
+    /// <summary>
+    /// Defines the contract for managing table-related operations.
+    /// </summary>
+    public interface ITableService
+    {
+        /// <summary>
+        /// Retrieves a list of all tables.
+        /// </summary>
+        /// <returns>A task that represents the asynchronous operation. 
+        /// The task result contains a list of table response DTOs.
+        /// </returns>
+        Task<IEnumerable<TableResponse>> GetAllAsync();
```

Comment:
you can just return `List` instead of `IEnumerable`, no actual benefit from returning that